The Docker build is failing in CI

In recent CI runs, we've been seeing Docker failures, which look like this:
```
+ ./version.sh
fatal: unsafe repository ('/workspace' is owned by someone else)
To add an exception for this directory, call:

	git config --global --add safe.directory /workspace
+ VERSION=
make: *** [build/package.BUILT] Error 128
```

I'm not familiar with how the Docker build works. Any help anyone can offer here would be appreciated!

--- a/docker_build.sh
+++ b/docker_build.sh
@@ -1,4 +1,6 @@
 #!/usr/bin/env sh
 set -e
+echo "Building the docker"
 docker build -t wasi-sdk-builder:latest .
+echo "Building the package in docker"
 docker run --mount type=bind,src=$PWD,target=/workspace -e NINJA_FLAGS=-v --workdir /workspace wasi-sdk-builder:latest make package

--- a/version.sh
+++ b/version.sh
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
 set -e
+git config --global --add safe.directory "/workspace"
 GIT_DESCR=$(git describe --long --candidates=999 --match='wasi-sdk-*' --dirty='+m' --abbrev=12)
 GIT_PACKAGE_VERSION=$(echo $GIT_DESCR | perl -ne 'if(/^wasi-sdk-(\d+)-(\d+)-g([0-9a-f]{7,12})([+]m)?$/) { if($2 == 0) { print "$1.$2$4" } else { print "$1.$2g$3$4" } exit } else { print "could not parse git description"; exit 1 }';)
 echo $GIT_PACKAGE_VERSION

